### PR TITLE
remove the option to pass in the --no-preview-dart-2 flag to analysis

### PR DESCRIPTION
**Pull request by devoncarew**
- remove the option to pass in the `--preview-dart-2` / `--no-preview-dart-2` to flutter analyze

@a-siva

### CODE REVIEW

**Comment by devoncarew** [comment]
@gspencergoog, I have not created a PR in the new cirrus world - are the failures on the analyze and test-linux bots expected?

File: packages/flutter_tools/test/commands/analyze_continuously_test.dart [hunk 1]
```diff
@@ -69,44 +69,6 @@ void main() {
...
-    });
-
-    await server.start();
-    await onDone;
-
-    expect(errorCount, 0);
-  }, overrides: <Type, Generator>{
-    OperatingSystemUtils: () => os
-  });
-
```
Comment (a-siva) [inline packages/flutter_tools/test/commands/analyze_continuously_test.dart]: why are you deleting the dart 2 mode for this test also?

File: packages/flutter_tools/test/commands/analyze_once_test.dart [hunk 2]
```diff
@@ -167,45 +167,6 @@ void bar() {
...
-        await runCommand(
-          command: new AnalyzeCommand(workingDirectory: fs.directory(tempDir)),
-          arguments: <String>['analyze', '--preview-dart-2'],
-          statusTextContains: <String>['No issues found!'],
-        );
-      } finally {
-        tempDir.deleteSync(recursive: true);
-      }
-    });
-
```
Comment (a-siva) [inline packages/flutter_tools/test/commands/analyze_once_test.dart]: Ditto comment about deleting the dart 2 mode test also...

**Review by a-siva — APPROVED** [review]
changes LGTM not sure why you are running into errors.

**Comment by gspencergoog** [comment]
The analyze error looks like a bug in the whitespace detection script: it's got:
```
grep: : No such file or directory
```
Where the two colons indicate that it's looking for a file with an empty string as the filename.

You can safely ignore that one as long as you don't have trailing whitespace on any of the lines.  I'll look at why it's trying to do that...

As for the Linux failure, it looks like Cirrus just terminated that one: they had a problem with that yesterday, and it should be fixed.  I restarted that test to see if it would finish.

**Comment by gspencergoog** [comment]
OK, I've looked into that analysis test, and I really don't see how it's failing the way it is: we explicitly filter out filenames that don't exist from the argument list before giving it to `grep`.

I thought maybe you were just a few commits behind on your master (before the fix to the whitespace check was committed), but you're up-to-date, so I'm left scratching my head... :-/

**Comment by gspencergoog** [comment]
Looking at your repo some more, I think you should update to the latest master HEAD and try again: I wasn't able to run `update-packages` on your `no_preview_dart_2_flag` branch as it was (it failed version checks because it couldn't figure out the version of Flutter), but once I rebased it locally on top of a current master, it worked fine.

File: packages/flutter_tools/test/commands/analyze_continuously_test.dart (hunk 1 above)
Comment (devoncarew) [reply packages/flutter_tools/test/commands/analyze_continuously_test.dart]: Good catch - I was a bit aggressive w/ deletes here. I restore one test in this file, and one in the file below.

**Comment by devoncarew** [comment]
@a-siva, thanks for the review, and @gspencergoog, thanks for taking a look at the build. I've updated the PR and merged to master; fingers crossed for a green build.

**Review by a-siva — APPROVED** [review]

**Comment by gspencergoog** [comment]
Looking good.

**Pull request merged**